src/main/scala/scalismo/statisticalmodel/StatisticalVolumeIntensityModel.scala

```diff
@@ -0,0 +1,100 @@
+package scalismo.statisticalmodel
+
+import breeze.linalg.DenseVector
+import scalismo.common._
+import scalismo.geometry._
+import scalismo.mesh._
+import scalismo.utils.Random
+
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe.TypeTag
+
+trait StatisticalVolumeIntensityModel[S] {
+
+  def referenceMesh: TetrahedralMesh3D
+
+  def referenceMeshField: ScalarVolumeMeshField[S]
+
+  def shape: StatisticalVolumeMeshModel
+
+  def intensity: DiscreteLowRankGaussianProcess[_3D, UnstructuredPointsDomain[_3D], S]
+
+  def landmarks: Map[String, Landmark[_3D]] = Map.empty[String, Landmark[_3D]]
+
+  def mean: ScalarVolumeMeshField[S]
+
+  def instance(coefficients: SVIMCoefficients): ScalarVolumeMeshField[S]
+
+  def sample()(implicit rnd: Random): ScalarVolumeMeshField[S]
+
+  def withLandmarks(landmarksMap: Map[String, Landmark[_3D]]): StatisticalVolumeIntensityModel[S]
+
+  def hasLandmarks: Boolean = landmarks.nonEmpty
+
+  def landmarkPointId(id: String): Option[PointId] = {
+    for {
+      lm <- landmarks.get(id)
+      id <- referenceMesh.pointSet.pointId(lm.point)
+    } yield id
+  }
+
+  def landmarksWithPointIds: Map[String, Option[PointId]] = landmarks.map { case (id, lm) => id -> referenceMesh.pointSet.pointId(lm.point) }
+
+  def landmarksWithClosestPointIds: Map[String, PointId] = landmarks.map { case (id, lm) => id -> referenceMesh.pointSet.findClosestPoint(lm.point).id }
+
+  def zeroCoefficients: SVIMCoefficients
+}
+
+object StatisticalVolumeIntensityModel {
+
+  def apply[S: Scalar: TypeTag: ClassTag](referenceMeshField: ScalarVolumeMeshField[S],
```

Comment:
It should be sufficient for the reference to be a TetrahedralMesh as it represents not an example function, but only the domain of the function.

Reply:
Doesn't the intensity model need a reference intensity,  even though it does not represent the domain? I though the reference intensities were subtracted from the targets ones i.e the deformation function values  are the differences between reference intensities and the target ones. In that case the reference intensity is needed as well (the reference should be a ScalarVolumeMeshField)

Reply:
It is important to distinguish between the reference and the mean. It needs a mean intensity, but as this is provided separately.

Reply:
I though as for shape deformation function where values are differences between reference points and target points which give vectors.

Reply:
In the case of the deformation model, the mean is indeed a difference from the reference shape to the mean shape. In case of the intensities, it could also be implemented like this. I am not sure, however, if this is not more complicated than just providing the mean intensity directly. 

What do you think?

Reply:
I think for consistency, we do the same as in shape model. I don't think it will be complicated.

Reply:
Lets consider as a concrete example that we specify a zero-mean Gaussian process. If we use the scalars of the reference mesh, this would mean that our assumptions are on average the same as the ones on the reference. If we only have a mesh on the reference, our model would say that the intensities would in average be zero. 

Which one is more intuitive?
I tend to agree with you that using a ScalarVolumeMesh as a reference, and only modelling the difference is better. What do others think?

Reply:
Considering your example, @rassaire's proposal makes sense. In that case the intensity model would be defined over vectors and not scalars?

Reply:
No, it can remain 1 dimensional. The difference here is just a scalar value (the difference to the mean intensity)

Reply:
You mean the difference to the reference?

Reply:
I updated the pull-request. The reference is again a  ScalarVolumeMeshField and intensities are generated by warping the reference intensities with the sampled differences.

Reply:
I agree using the reference image as a reference, the sampled intensity from such a model will be close to the original intensity range of the imaging modality, which is not always the case for the mean due to the correspondence accuracy.